### ISSUE TITLE
how to check the hd5 data ?

### ISSUE THREAD

**Issue by wuqiangch** [luoyetx/deep-landmark#6]
After training by using bootstrap.sh, I get the wrong result .
![4febec6cd3fa9e75d2a28023c0b0bc76](https://cloud.githubusercontent.com/assets/7983007/13692196/5870bc9a-e77a-11e5-8d9d-4b997b4fae73.jpg)
![b656dd97cb9670160ad455223cd61a6c](https://cloud.githubusercontent.com/assets/7983007/13692197/5b613588-e77a-11e5-8149-d5dddf0a064c.jpg)

I use HDFView the 1_F/train.hd5 file,what  are the meaning of data and landmark?
(1)landmark,first six rows:
0.27142859  0.27142859  0.6571429   0.27142859  0.3 0.6142857   0.31428573  0.78571427  0.62857145  0.78571427
0.32422328  0.1958884   0.8004285   0.24294785  0.49501193  0.4644839   0.2713394   0.6769809   0.68675554  0.74030954
0.21656966  0.22970417  0.7015257   0.23771922  0.58244133  0.60628456  0.28283 0.8140927   0.68875426  0.7836232
0.2470238   0.22058824  0.6875  0.24411765  0.39583334  0.5029412   0.24107143  0.7617647   0.6279762   0.7852941
0.26837537  0.24787492  0.765999    0.2968669   0.5151268   0.6077065   0.19847062  0.6727469   0.6902065   0.7266793
0.4111675   0.30964467  0.7817259   0.20812182  0.75126904  0.6040609   0.5329949   0.8071066   0.78680205  0.7360406
(2)data's six rows
1.5018264
-1.6148947
-1.7687876
-1.742761
-2.0221322
-1.746762
“data” only has a  column and "landmark" has 10 columns . Is it right?

**Comment by luoyetx** [comment]
@wuqiangch the result here is 3 level ? and the level-1 includes EN and NM ?

what's more, did you monitor the loss during network training? Maybe some cnns are not trained well.

**Comment by luoyetx** [comment]
the training data you have put out is ok. 5 landmark has 10 values and image data is a multi-dimension variable so it may be hard to view all values.

you may refer to this [pdf](https://github.com/luoyetx/face-alignment-presentation/blob/master/Face-Alignment.pdf), in latter half part, I put some more detailed data about the training status.

**Comment by wuqiangch** [comment]
@luoyetx yes,I used my trained 23 models.But two models(1_EN,1_NM)  was not be used. I used my 21 models to repace your models under webapp/v0.0.6 and test it. I used my 8 k80 gpus to train together. Its error is very low.

**Comment by wuqiangch** [comment]
@luoyetx  I can't open your pdf. So if  it is possible,please send it to my email (wuqiang_jobs@163.com).Thanks!

**Comment by luoyetx** [comment]
the model I provided under webapp/v0.0.6, did it have the same problem?

**Comment by wuqiangch** [comment]
@luoyetx  no, it is ok.

**Comment by wuqiangch** [comment]
@luoyetx  I have another question. Caffe with different version will get different train.hd5.
Why?

**Comment by luoyetx** [comment]
the pdf is on github, the loading may cost some time but should be able to view it, if not, you can just download directly fro github

the training data is processed by Python code which has nothing to do with Caffe

**Comment by wuqiangch** [comment]
This  wrong result  may be caused by caffe with diffrent version. You can upload your caffe to github.

**Comment by luoyetx** [comment]
I don't think Caffe is the problem, the Caffe version I use is about 10 months ago. I already test the model with recently Caffe, it's OK.

What you mean here may be the training part is not  compact with different Caffe version? I think it's impossible.

Maybe you should pay more attention to the training status of 21 networks to see if the loss is small enough and stable.

**Comment by luoyetx** [comment]
By the way, you can test your model by

```
python test/test.py 0 # to test 1_F
python test/test.py 2 # to test 1_F and level2
python test/test.py 3 # to test 1_F and level2 and level3
```

you can also modify the webapp code to view different result of different level

**Comment by wuqiangch** [comment]
I used bootstrap.sh to train it and i got 23 models

**Comment by luoyetx** [comment]
@wuqiangch  I recommend you to train the models manually using Caffe command which you can control the training status. The models I provide are well tuned which may not use the default parameters in prototxt files.

**Comment by wuqiangch** [comment]
There are my results:
LEVEL - 0
Mean Error:
    Left Eye       = 0.022836
    Right Eye      = 0.253351
    Nose           = 0.029818
    Left Mouth     = 0.028378
    Right Mouth    = 0.028626
Failure:
    Left Eye       = 0.061743
    Right Eye      = 1.000000
    Nose           = 0.138488
    Left Mouth     = 0.108194
    Right Mouth    = 0.118003
![f_test](https://cloud.githubusercontent.com/assets/7983007/13695339/a1cd5a7e-e798-11e5-9c78-04e707c6ea2a.png)

LEVEL - 2
Mean Error:
    Left Eye       = 0.015997
    Right Eye      = 0.336471
    Nose           = 0.018802
    Left Mouth     = 0.128772
    Right Mouth    = 0.020509
Failure:
    Left Eye       = 0.033468
    Right Eye      = 1.000000
    Nose           = 0.038950
    Left Mouth     = 0.997403
    Right Mouth    = 0.058857

LEVEL - 3
Mean Error:
    Left Eye       = 0.062053
    Right Eye      = 0.337876
    Nose           = 0.018445
    Left Mouth     = 0.175577
    Right Mouth    = 0.019996
Failure:
    Left Eye       = 0.905366
    Right Eye      = 1.000000
    Nose           = 0.037507
    Left Mouth     = 0.999711
    Right Mouth    = 0.056261
![Uploading level3_test.png…]()

**Comment by luoyetx** [comment]
It seems 1_F's right eye is wrong, the error is too large, what's more in level-2,  left mouth's error is too large, you should retrain related cnns.

**Comment by wuqiangch** [comment]
How to modify my prototxts?

**Comment by luoyetx** [comment]
I think re-train 1_F and 2_LM1, 2_LM2 will give you a good result if the loss of these network is pretty low and stable. There's no need to modify the prototxts only if you want a better result.

**Comment by wuqiangch** [comment]
Thanks,I will try it.
